core/server/utils/request.js

```diff
@@ -0,0 +1,29 @@
+var got = require('got'),
+    Promise = require('bluebird'),
+    validator = require('../data/validation').validator,
+    errors = require('../errors'),
+    _  = require('lodash');
+
+module.exports = function request(url, options) {
+    if (_.isEmpty(url) || !validator.isURL(url)) {
+        return Promise.reject(new errors.InternalServerError({
+            message: 'URL empty or invalid.',
+            code: 'GOT_REQUEST',
+            context: url
+        }));
+    }
+
+    return got(
+        url,
+        options
+    ).then(function (response) {
+        return Promise.resolve(response);
+    }).catch(function (err) {
+        return Promise.reject(new errors.InternalServerError({
```

Comment:
IMO this should just be `Promise.reject(err)`, else the calling library won't be able to detect timeouts etc

Reply:
Makes sense to me! I was using the `statusCode` and `statusMessage` but it would actually be better to let the caller deal with this. Will push the changes quickly.